include/extractor/guidance/is_through_street.hpp

```diff
@@ -0,0 +1,61 @@
+#include "extractor/guidance/sliproad_handler.hpp"
+#include "extractor/guidance/constants.hpp"
+#include "util/assert.hpp"
+#include "util/bearing.hpp"
+#include "util/coordinate_calculation.hpp"
+#include "util/guidance/name_announcements.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <iterator>
+#include <limits>
+
+#include <boost/assert.hpp>
+
+using osrm::extractor::guidance::getTurnDirection;
+using osrm::util::angularDeviation;
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+// template <typename restriction_type> auto asDuplicatedNode(const restriction_type &restriction)
+// {
+//     auto &way = restriction.AsWayRestriction();
+//     // group restrictions by the via-way. On same via-ways group by from
+//     return std::tie(way.in_restriction.via, way.out_restriction.via, way.in_restriction.from);
+// }
+
+bool isThroughStreet(const EdgeID from, const IntersectionView &intersection, const util::NodeBasedDynamicGraph &node_based_graph, const EdgeBasedNodeDataContainer &node_data_container, const util::NameTable &name_table, const SuffixTable &street_name_suffix_table)
```

Comment:
Needs to be inline otherwise we have multiple function definitions violating the One-Definition Rule (ODR)